physlr/physlr.py

```diff
@@ -1276,6 +1287,25 @@ def physlr_mst_troubleshoot(self):
         self.write_graph(gmst, sys.stdout, self.args.graph_format)
         print(int(timeit.default_timer() - t0), "Wrote the MST.", file=sys.stderr)
 
+    def physlr_report_junctions_graph(self):
+        """
+        Report junctions in the MST of the graph and output a list of junc barcodes.
+        """
+        g = self.read_graph(self.args.FILES)
+        gmst = Physlr.determine_pruned_mst(g)
+        print(int(timeit.default_timer() - t0), "Searching for junctions...", file=sys.stderr)
+        junctions = []
+        for component in nx.connected_components(gmst):
+            junctions += Physlr.detect_junctions_of_tree(
+                gmst.subgraph(component), Physlr.args.prune_junctions)
+        print(int(timeit.default_timer() - t0),
+              "Found", len(junctions), "junctions.", file=sys.stderr)
+        gjunctions = nx.Graph()
+        gjunctions.add_nodes_from(junctions)
```

Comment:
Just wondering -- is it necessary to format this output as a graph? It looks like you are just adding nodes, but no edges, so couldn't you just use a TSV file of the junction nodes?

Reply:
Right, I could. I found it easier at that time to do it that way as the method `write_tsv()` writes graphs in tsv format. I'll add a few lines of code to write the list as tsv.